src/components/interactive-builder/add-question.modal.tsx

```diff
@@ -79,6 +73,21 @@ interface RequiredLabelProps {
   t: TFunction;
 }
 
+export const getDatePickerType = (concept: Concept): DatePickerType | null => {
+  const conceptDataType = concept.datatype.name;
+  switch (conceptDataType) {
+    case 'Datetime':
+      return 'both';
+    case 'Date':
+      return 'calendar';
+    case 'Time':
+      return 'timer';
+    default:
+      console.warn(`Unsupported datatype for date fields: ${conceptDataType}`);
+      return null;
+  }
+};
```

Comment:
To prevent user's from selecting a date picker of type say `both` that has both calendar and timer when the concept is only of type `date` and the time part would get cut off in the backend because of the concept's data type.

Reply:
In summary, the kind of date (date vs date-time vs time) that one submits should match with the concept's datatype.

Reply:
Cool!